### PR TITLE
Faster reader query (#146)

### PR DESCRIPTION
**Pull request by holtgrewe**
This also includes the "Also recognize "bgz" suffix as bgziped for indexed fasta reader" patch...

### CODE REVIEW

File: noodles-fasta/src/reader.rs [hunk 1]
```diff
@@ -306,6 +310,57 @@ where
...
+        let reader_buf = reader.fill_buf()?;
+
+        if reader_buf.is_empty() || reader_buf[0] == DEFINITION_PREFIX {
+            break;
+        }
+
+        let len = match memchr(LINE_FEED, reader_buf) {
+            Some(i) => {
+                let i = std::cmp::min(i, bases_left);
+                bases_left -= i;
```
Comment (zaeleus) [inline noodles-fasta/src/reader.rs:336]: This doesn't compensate for `CRLF` newlines. E.g., `AC\r\nGT` would return `i = 3` when there are 2 bases.

Comment (holtgrewe) [reply noodles-fasta/src/reader.rs]: Thanks, good catch. Please consider my change, also reproduced below for convenience.

```diff
diff --git a/noodles-fasta/src/reader.rs b/noodles-fasta/src/reader.rs
index 12acdfb2..10e2b067 100644
--- a/noodles-fasta/src/reader.rs
+++ b/noodles-fasta/src/reader.rs
@@ -333,14 +333,15 @@ where
         let len = match memchr(LINE_FEED, reader_buf) {
             Some(i) => {
                 let i = std::cmp::min(i, bases_left);
-                bases_left -= i;
                 let line = &reader_buf[..i];

                 if line.ends_with(&[CARRIAGE_RETURN]) {
                     let end = line.len() - 1;
                     buf.extend(&line[..end]);
+                    bases_left -= end;
                 } else {
                     buf.extend(line);
+                    bases_left -= i;
                 }

                 i + 1
```

**Comment by holtgrewe** [comment]
@zaeleus could you make a new release of the `noodles-fasta` crate? I would like to depend on this in [another crate](https://github.com/bihealth/seqrepo-rs).

**Comment by zaeleus** [comment]
noodles 0.33.0 / noodles-fasta 0.19.0 is now published.

**Comment by holtgrewe** [comment]
🙇

**Pull request merged**